src/main/java/com/getindata/connectors/http/internal/table/lookup/BodyBasedRequestFactory.java

```diff
@@ -53,9 +53,15 @@ protected Logger getLogger() {
         return log;
     }
 
-    private URI constructGetUri() {
+    URI constructBodyBasedUri(LookupQueryInfo lookupQueryInfo) {
+        StringBuilder resolvedUrl = new StringBuilder(baseUrl);
+        if (lookupQueryInfo.hasBodyBasedUrlQueryParameters()) {
+            resolvedUrl.append(baseUrl.contains("?") ? "&" : "?")
+                       .append(lookupQueryInfo.getBodyBasedUrlQueryParameters());
+        }
```

Comment:
I'm guessing you are planing to have your own `LookupQueryCreator` custom implementation that will return a `LookupQueryInfo` that will have both, query and body params right?

And you plan to use it with one of the `HttpRequestFactory` implementation right?

I wonder about contract between  `LookupQueryInfo` and `HttpRequestFactory`.
In other words I wonder if the logic above is an implementation detail of `LookupQueryInfo` or public contract that `LookupQueryInfo` creator should know.